core/src/main/scala-3/spire/syntax/Syntax.scala

```diff
@@ -0,0 +1,729 @@
+    def equotmod(rhs: Int)(using ev1: Ring[A]): (A, A) = er.equotmod(lhs, ev1.fromInt(rhs))
+    //
+    def equot(rhs: Double)(using ev1: Field[A]): A = er.equot(lhs, ev1.fromDouble(rhs))
+    def emod(rhs: Double)(using ev1: Field[A]): A = er.emod(lhs, ev1.fromDouble(rhs))
+    def equotmod(rhs: Double)(using ev1: Field[A]): (A, A) = er.equotmod(lhs, ev1.fromDouble(rhs))
+
+    /* TODO: move to TruncatedDivision
+    def /~(rhs:Number)(using c:ConvertableFrom[A]): Number = c.toNumber(lhs) /~ rhs
+    def %(rhs:Number)(using c:ConvertableFrom[A]): Number = c.toNumber(lhs) % rhs
+    def /%(rhs:Number)(using c:ConvertableFrom[A]): (Number, Number) = c.toNumber(lhs) /% rhs
+    */
+  extension(lhs: Int)
+    def equot[A](rhs: A)(using ev: EuclideanRing[A]): A = ev.equot(ev.fromInt(lhs), rhs)
+    def emod[A](rhs: A)(using ev: EuclideanRing[A]): A = ev.emod(ev.fromInt(lhs), rhs)
+    def equotmod[A](rhs: A)(using ev: EuclideanRing[A]): (A, A) = ev.equotmod(ev.fromInt(lhs), rhs)
+
+  extension(lhs: Long)
+    def equot[A](rhs: A)(using ev: EuclideanRing[A], c: ConvertableTo[A]): A = ev.equot(c.fromLong(lhs), rhs)
+    def emod[A](rhs: A)(using ev: EuclideanRing[A], c: ConvertableTo[A]): A = ev.emod(c.fromLong(lhs), rhs)
+    def equotmod[A](rhs: A)(using ev: EuclideanRing[A], c: ConvertableTo[A]): (A, A) =
+      ev.equotmod(c.fromLong(lhs), rhs)
+
+  extension(lhs: Double)
+    def equot[A](rhs: A)(using ev: Field[A]): A = ev.equot(ev.fromDouble(lhs), rhs)
+    def emod[A](rhs: A)(using ev: Field[A]): A = ev.emod(ev.fromDouble(lhs), rhs)
+    def equotmod[A](rhs: A)(using ev: Field[A]): (A, A) = ev.equotmod(ev.fromDouble(lhs), rhs)
+
+trait FieldSyntax extends EuclideanRingSyntax with MultiplicativeGroupSyntax
+
+trait NRootSyntax {
+  extension [A](lhs: A)(using ev: NRoot[A])
+    def nroot(rhs: Int): A = ev.nroot(lhs, rhs)
+    def sqrt(): A = ev.sqrt(lhs)
+    def fpow(rhs: A): A = ev.fpow(lhs, rhs)
+
+    def pow(rhs: Double)(using c: Field[A]): A = ev.fpow(lhs, c.fromDouble(rhs))
+    def **(rhs: Double)(using c: Field[A]): A = ev.fpow(lhs, c.fromDouble(rhs))
+
+    def pow(rhs: Number)(using c: ConvertableFrom[A]): Number = c.toNumber(lhs).pow(rhs)
+    def **(rhs: Number)(using c: ConvertableFrom[A]): Number = c.toNumber(lhs) ** rhs
+
+  extension(lhs: Int)
+    def **[A](rhs: A)(using ev: NRoot[A], c: ConvertableTo[A]): A = ev.fpow(c.fromInt(lhs), rhs)
+
+  extension(lhs: Long)
+    def **[A](rhs: A)(using ev: NRoot[A], c: ConvertableTo[A]): A = ev.fpow(c.fromLong(lhs), rhs)
+
+  extension(lhs: Double)
+    def **[A](rhs: A)(using ev: NRoot[A], c: ConvertableTo[A]): A = ev.fpow(c.fromDouble(lhs), rhs)
+}
+
+trait LeftModuleSyntax extends RingSyntax:
+  implicit def lms[V](v: V): LeftModuleOps[V] = new LeftModuleOps[V](v)
+  // Left module
+  // extension[V](x: V)
+  //   def *:[F](lhs: F)(using ev: LeftModule[V, F]): V = ev.timesl(lhs, x)
+    // def *:[F](lhs: Int)(using ev: LeftModule[V, F], F: Ring[F]): V = ev.timesl(F.fromInt(lhs), x)
```

Comment:
Right, I (maybe naively) started converting all the syntax to extension methods. Only later I realized that `spire` uses implicits inheritance to establish a hirerachy or prioritization which cannot (Apparently) be made with extension methods. I could solve most of them except a few (3 I think) that may need some extra work.

An alternative is to go back to syntax via implicit conversions rather than extensions

Reply:
I think initially, the implicit conversion approach should be preferred.

Reply:
Yes, I agree, can we go back to that?